lib/jormungandr/test/bench/Latency.hs

```diff
@@ -294,68 +377,82 @@ isLogRequestFinish = \case
+    LogText txt -> (T.isInfixOf "MsgObserveStart" txt)
+    _ -> False
+
+isLogObserveStop :: ServerLog -> Bool
+isLogObserveStop = \case
+    LogText txt -> (T.isInfixOf "MsgObserveStop" txt)
+    _ -> False
+
 measureApiLogs
     :: TVar [LogObject ServerLog] -- ^ Log message variable.
     -> IO a -- ^ Action to run
-    -> IO [NominalDiffTime]
-measureApiLogs = measureLatency isLogRequestStart isLogRequestFinish
+    -> IO ([NominalDiffTime],[NominalDiffTime])
+measureApiLogs = measureLatency
+    isLogRequestStart
+    isLogRequestFinish
+    isLogObserveStart
+    isLogObserveStop
 
 -- | Run tests for at least this long to get accurate timings.
 sampleTimeSeconds :: Int
-sampleTimeSeconds = 5
+sampleTimeSeconds = 1
+
+-- | Run tests for at least this long to get accurate timings.
+sampleNTimes :: Int
+sampleNTimes = 10
 
 -- | Measure how long an action takes based on trace points and taking an
--- average of results over a short time period.
+-- average of results over a short time period (also DB footprint).
 measureLatency
     :: (msg -> Bool) -- ^ Predicate for start message
     -> (msg -> Bool) -- ^ Predicate for end message
+    -> (msg -> Bool) -- ^ Predicate for start DB observe message
+    -> (msg -> Bool) -- ^ Predicate for stop DB observe message
     -> TVar [LogObject msg] -- ^ Log message variable.
     -> IO a -- ^ Action to run
-    -> IO [NominalDiffTime]
-measureLatency start finish tvar action = do
+    -> IO ([NominalDiffTime], [NominalDiffTime])
+measureLatency start finish startDB stopDB tvar action = do
     atomically $ writeTVar tvar []
-    _res <- repeatFor sampleTimeSeconds action
-    extractTimings start finish . reverse <$> readTVarIO tvar
+    replicateM_ sampleNTimes action
+    extractTimings start finish startDB stopDB . reverse <$> readTVarIO tvar
 
 -- | Scan through iohk-monitoring logs and extract time differences between
--- start and end messages.
+-- start/finish and also DBLog's related msgs that allow compute DB footprint.
 extractTimings
     :: (a -> Bool) -- ^ Predicate for start message
     -> (a -> Bool) -- ^ Predicate for end message
+    -> (a -> Bool) -- ^ Predicate for start DB observe message
+    -> (a -> Bool) -- ^ Predicate for stop DB observe message
     -> [LogObject a] -- ^ Log messages
-    -> [NominalDiffTime]
-extractTimings isStart isFinish msgs = map2 mkDiff filtered
+    -> ([NominalDiffTime],[NominalDiffTime])
+extractTimings isStart isFinish isObserveStart isObserveStop msgs =
+    (map2 mkDiff filtered, map2 mkDiff filteredDB)
   where
     map2 _ [] = []
     map2 f (a:b:xs) = (f a b:map2 f xs)
-    map2 _ _ = error "start trace without matching finish trace"
+    map2 _ _ = []
 
     mkDiff (False, start) (True, finish) = diffUTCTime finish start
-    mkDiff (False, _) _ = error "missing finish trace"
-    mkDiff (True, _) _ = error "missing start trace"
+    mkDiff (False, start) _ = diffUTCTime start start
+    mkDiff (True, finish) _ = diffUTCTime finish finish
```

Comment:
probably should not do that, I see DBlogs not coming with pairs and they should! need to look at this again